### PR TITLE
Profile model extra props and profile edit view

### PR DESCRIPTION
**Pull request by Braagaa**
**What it does:**

- Extra props are needed in the Profile model in order to search and filter sitters. These are ratings, hourly rate, and an extra prop job title
- Job title will be a list of jobs the sitter is more inclined too such as dog trainer, dog sitter, dog walker, etc
- Profile Edit view also has a condition for when it is a sitter. It will get 2 extra fields. One is to input their hourly rate and the second is to select their job titile.
- Update Profile Edit now uses @dproc96 Profile Context

**UPDATE**

- Added extra prop "address" to Profle Model for friendlier readable address
- Added MapQuest geocode api to help standardize addresses and to obtain GPS Points for the database

### CODE REVIEW

File: client/src/components/ProfileEdit/data.js [hunk 1]
```diff
@@ -5,3 +5,11 @@ export const genders = [
   "Other",
   "Prefer Not To Say",
 ];
+
+export const jobTitles = [
```
Comment (dproc96) [inline client/src/components/ProfileEdit/data.js:9]: Interesting addition to the app, I like this idea for expanding the market

File: client/src/components/ProfileEdit/index.js [hunk 2]
```diff
@@ -194,6 +157,8 @@ const ProfileEdit = function () {
       },
       body: JSON.stringify(cleanForm),
     })
+      .then((res) => res.json())
+      .then(({ profile }) => setProfile(profile))
```
Comment (dproc96) [inline client/src/components/ProfileEdit/index.js]: This is a much better way of doing what I did in my PR for the Dog Sitter switch. Once this is merged we can use this instead!

**Review by dproc96 — APPROVED** [review]
This looks good to me! Nice additions

File: client/src/components/ProfileEdit/index.js [hunk 3]
```diff
@@ -131,47 +94,47 @@ const ProfileEdit = function () {
...
-      setErrors({ ...errors, email: "" });
+      setErrors({ ...errors, phone: "" });
       setDisableSubmit(false);
     }
 
-    setValues({ ...values, email: text });
+    setValues({ ...values, phone: value });
   };
 
+  const handleEmail = regExHandler(/.+@.+..+/, "Email is not valid", "email");
```
Comment (dahrk) [inline client/src/components/ProfileEdit/index.js]: This regex evaluator misses a few cases, i.e. tony@test passes the validation. There are some more robust regex checks for email online that we can refer from

File: client/src/components/ProfileEdit/index.js [hunk 4]
```diff
@@ -131,47 +94,47 @@ const ProfileEdit = function () {
...
+  const handleHourlyRate = regExHandler(
+    /^\d*(\.\d{0,2})?$/,
```
Comment (dahrk) [inline client/src/components/ProfileEdit/index.js:131]: 29123. will pass this check, might be good to append 00 to the end as necessary

File: client/src/components/ProfileEdit/index.js [hunk 5]
```diff
@@ -194,6 +157,8 @@ const ProfileEdit = function () {
       },
       body: JSON.stringify(cleanForm),
     })
+      .then((res) => res.json())
+      .then(({ profile }) => setProfile(profile))
       .then(() => setIsSaved(true));
```
Comment (dahrk) [inline client/src/components/ProfileEdit/index.js:170]: You'll likely have to update the user.context as well. A changed email does not show on the front end until the app is refreshed

File: client/src/components/ProfileEdit/index.js [hunk 6]
```diff
@@ -131,47 +94,47 @@ const ProfileEdit = function () {
...
 
-    setValues({ ...values, phone: value });
+    setValues({ ...values, [prop]: value });
   };
 
-  const handleEmail = (e) => {
-    const text = e.currentTarget.value;
-    if (!/.+@.+..+/.test(text)) {
-      setErrors({ ...errors, email: "Email is not valid" });
+  const handlePhone = (value) => {
```
Comment (dahrk) [inline client/src/components/ProfileEdit/index.js]: Is it possible to make this also work with regExHandler?

File: client/src/components/ProfileEdit/index.js [hunk 7]
```diff
@@ -260,29 +226,47 @@ const ProfileEdit = function () {
...
-            variant="outlined"
-          >
-            {genders.map((value) => (
-              <MenuItem value={value} key={value}>
-                {value}
-              </MenuItem>
-            ))}
-          </TextField>
+      {/*HOURLY RATE*/}
+      {profile && profile.isSitter && (
```
Comment (dahrk) [inline client/src/components/ProfileEdit/index.js:253]: This and l252 can probably be wrapped together in case anything chances with the variables

**Review by dahrk — CHANGES_REQUESTED** [review]
Lets make sure that email is updated in user.context before merging this one

**Review by dahrk — APPROVED** [review]
Functionality is working but the profileContext is not being updated on editProfile until the page is refreshed. At a glance you may have to use setProfile(cleanForm) instead of setProfile(profile)

Approving because you need this for another branch, make the profile change before merge

**Pull request merged**